src/Common/ItechArt.Common/OperationResult.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System;
+
+namespace ItechArt.Common;
+
+public class OperationResult<TResult, TError>
+    where TError : Enum
```

Comment:
1. let's add struct constraint
2. change `Error` propery type to nullable (get rid of _error field)